frontend/src/metabase/parameters/components/ParameterSidebar/ParameterSidebar.tsx

```diff
@@ -130,6 +135,7 @@ const ParameterSidebar = ({
         {tab === "settings" ? (
           <ParameterSettings
             parameter={parameter}
+            otherParameterSlugs={otherParameterSlugs}
```

Comment:
looks wrong to me from the composition perspective. can't we decide on current level if the value is used for another component?

Reply:
Can you be more specific on how the alternative would look? `ParameterSettings` needs to be able to know whether the slug in its input is already used for another parameter. I can't see how else it can do that, except this implementation, or lifting up the state of its input to be in the ParameterSidebar component, which is clearly worse to me.